### PR TITLE
[WIP]Refactoring ErrorList into NuGet.VisualStudio.Common

### PR DESCRIPTION
**Pull request by mishra14**
Follow up to https://github.com/NuGet/NuGet.Client/pull/2313

As requested by @nkolev92, I have refactored the code to move `ErrorListTableEntry.cs` and `ErrorListTableDataSource.cs` from `NuGet.SolutionRestoreManager` to `NuGet.VisualStudio.Common` and removed the dependency from `NuGet.Tools` to `NuGet.SolutionRestoreManager`

### CODE REVIEW

File: NuGet.sln [hunk 1]
```diff
@@ -222,12 +222,14 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "NuGet.Shared.Tests", "test\
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "NuGet.MSSigning.Extensions.FuncTest", "test\NuGet.Clients.FuncTests\NuGet.MSSigning.Extensions.FuncTest\NuGet.MSSigning.Extensions.FuncTest.csproj", "{BA7F681E-FDC7-42B6-8BF1-74DA6421BA1A}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "NuGet.MSSigning.Extensions.Test", "test\NuGet.Clients.Tests\NuGet.MSSigning.Extensions.Test\NuGet.MSSigning.Extensions.Test.csproj", "{2A604A43-C72A-4A03-B68B-66AF4B6F430E}"
```
Comment (nkolev92) [inline NuGet.sln:225]: Why is this changing?

File: src/NuGet.Clients/NuGet.Tools/NuGet.Tools.csproj [hunk 2]
```diff
@@ -106,10 +106,6 @@
       <Project>{538adefd-2170-40a9-a2c5-ec8369cfe490}</Project>
       <Name>NuGet.PackageManagement.UI</Name>
     </ProjectReference>
-    <ProjectReference Include="..\NuGet.SolutionRestoreManager\NuGet.SolutionRestoreManager.csproj">
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.Tools/NuGet.Tools.csproj:109]: 👏

**Review by nkolev92 — APPROVED** [review]
LGTM

**Comment by mishra14** [comment]
closing in favor of https://github.com/NuGet/NuGet.Client/pull/2329

**Pull request closed without merging**